opal/corelib/string/encoding.rb

```diff
@@ -232,6 +232,13 @@ class String
   `Opal.defineProperty(String.prototype, 'internal_encoding', #{Encoding::UTF_8})`
 
   def bytes
+    # REMIND: required when running in strict mode, otherwise the following error will be thrown:
+    # Cannot create property 'bytes' on string 'abc'
+    %x{
+      if (typeof self === 'string') {
+        self = new String(self)
```

Comment:
should we call `self.$dup()`?

Reply:
thoughts on just not memoizing the bytes array?

e.g.

```rb
    %x{
      if (typeof self === 'string') {
        return #{`new String(self)`.each_byte.to_a};
      }
    }
```

Also do we really need to memoize it?

Reply:
Probably not... sure it will be faster if we call `#bytes` on a loop but I doubt it's an actual/legit use case.